sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+    private QualityGate.Operator operator;
+    private String errorThreshold;
+    private String warningThreshold;
+    private boolean onLeakPeriod;
+
+    private ConditionBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public ConditionBuilder setMetricKey(String metricKey) {
+      this.metricKey = requireNonNull(metricKey, METRIC_KEY_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ConditionBuilder setOperator(QualityGate.Operator operator) {
+      this.operator = requireNonNull(operator, OPERATOR_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ConditionBuilder setErrorThreshold(@Nullable String errorThreshold) {
+      this.errorThreshold = errorThreshold;
+      return this;
+    }
+
+    public ConditionBuilder setWarningThreshold(@Nullable String warningThreshold) {
+      this.warningThreshold = warningThreshold;
+      return this;
+    }
+
+    public ConditionBuilder setOnLeakPeriod(boolean onLeakPeriod) {
+      this.onLeakPeriod = onLeakPeriod;
+      return this;
+    }
+
+    public QualityGate.Condition buildNoValue() {
+      checkCommonProperties();
+      return new QualityGate.Condition() {
+        @Override
+        public QualityGate.EvaluationStatus getStatus() {
+          return QualityGate.EvaluationStatus.NO_VALUE;
+        }
+
+        @Override
+        public String getMetricKey() {
+          return metricKey;
+        }
+
+        @Override
+        public QualityGate.Operator getOperator() {
+          return operator;
+        }
+
+        @Override
+        public String getErrorThreshold() {
+          return errorThreshold;
+        }
+
+        @Override
+        public String getWarningThreshold() {
+          return warningThreshold;
+        }
+
+        @Override
+        public boolean isOnLeakPeriod() {
+          return onLeakPeriod;
+        }
+
+        @Override
+        public String getValue() {
+          throw new IllegalStateException("There is no value when status is NO_VALUE");
+        }
+
+        @Override
+        public String toString() {
+          return "Condition{" +
+            "status=" + QualityGate.EvaluationStatus.NO_VALUE +
+            ", metricKey='" + metricKey + '\'' +
+            ", operator=" + operator +
+            ", errorThreshold='" + errorThreshold + '\'' +
+            ", warningThreshold='" + warningThreshold + '\'' +
+            ", onLeakPeriod=" + onLeakPeriod +
+            '}';
+        }
+      };
+    }
+
+    public QualityGate.Condition build(final QualityGate.EvaluationStatus status, final String value) {
+      checkCommonProperties();
+      requireNonNull(status, "status can not be null");
+      checkArgument(status != QualityGate.EvaluationStatus.NO_VALUE, "status can not be NO_VALUE, use method buildNoValue() instead");
+      requireNonNull(value, "value can not be null, use method buildNoValue() instead");
+      return new QualityGate.Condition() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 49 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)